model/model_training/models/rope.py

```diff
@@ -0,0 +1,175 @@
+import torch
+
+
+# rotary pos emb helpers (torch.jit.script does not seem to support staticmethod...)
+def rotate_half(x):
+    x1, x2 = x[..., : x.shape[-1] // 2], x[..., x.shape[-1] // 2 :]
+    return torch.cat((-x2, x1), dim=x1.ndim - 1)  # dim=-1 triggers a bug in torch < 1.8.0
+
+
+class RWNTKScaledRope(torch.nn.Module):
```

Comment:
Could you add some comments about the origin of these rope scaling variants? Were they copied from the [scaled-rope](https://github.com/jquesnelle/scaled-rope) repository? If they were imported from a different place we should add a proper attribution.